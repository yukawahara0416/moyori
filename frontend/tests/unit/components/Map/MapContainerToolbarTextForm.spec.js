import { mount } from '@vue/test-utils'
import Component from '@/components/Map/MapContainerToolbarTextForm.vue'

let wrapper

beforeEach(() => {
  wrapper = mount(Component, {})
})

afterEach(() => {
  wrapper.destroy()
})

describe('v-on', () => {
  it('textSearch', () => {
    const event = jest.fn()
    wrapper.setMethods({ textSearch: event })
    wrapper.find('input').trigger('keydown.enter')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('emit', () => {
  it('textSearch', () => {
    wrapper.vm.textSearch()
    expect(wrapper.emitted('text-search')).toBeTruthy()
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
