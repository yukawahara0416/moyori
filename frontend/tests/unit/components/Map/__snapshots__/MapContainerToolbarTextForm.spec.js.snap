// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div>
  <div
    class="v-input mx-2 v-input--hide-details theme--light v-text-field v-text-field--single-line"
  >
    <div
      class="v-input__prepend-outer"
    >
      <div
        class="v-input__icon v-input__icon--prepend"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-magnify theme--light"
        />
      </div>
    </div>
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <label
            class="v-label theme--light"
            for="input-12"
            style="right: 0px; position: absolute;"
          >
            キーワード検索
          </label>
          <input
            id="input-12"
            type="text"
          />
        </div>
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--clear"
          >
            <button
              aria-label="clear icon"
              class="v-icon notranslate v-icon--disabled v-icon--link material-icons theme--light"
              disabled="disabled"
              type="button"
            >
              $clear
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
